litho-processor/src/main/java/com/facebook/litho/specmodels/generator/EventGenerator.java

```diff
@@ -119,16 +122,74 @@ static TypeSpecDataHolder generateEventDispatcher(EventDeclarationModel eventDec
             .addModifiers(Modifier.STATIC)
             .addParameter(ClassNames.EVENT_HANDLER, "_eventHandler");
 
-    eventDispatcherMethod.addStatement(
-        "final $T _eventState = new $T()", eventDeclaration.name, eventDeclaration.name);
+    final TypeElement eventRepresentedObject = (TypeElement) eventDeclaration.representedObject;
+    final boolean eventClassHasOnlyOneNonEmptyCtor =
+        TypeElementUtils.hasOnlyOneNonEmptyConstructor(eventRepresentedObject);
+
+    /*
+    * Check if our class has any non-empty constructors. If so we will instantiate the
+    * Event class using that one instead of accessing mutable fields.
+    * */
+    if (eventClassHasOnlyOneNonEmptyCtor) {
+      final StringBuilder ctorParamsPlaceholder = new StringBuilder();
+      final List<String> ctorParamNames = new ArrayList<>();
+
+      /*
+      * Iterate over the Event fields in order to achieve the following:
+      *
+      * 1) Add the parameters to the method under construction.
+      * 2) Create a parameter name placeholder for JavaPoet.
+      * 3) Keep a list of all parameter names.
+      * */
+      for (FieldModel fieldModel : eventDeclaration.fields) {
+        eventDispatcherMethod
```

Comment:
We should skip `static` (`static final`) fields here as well.

Reply:
Sure will update it!